### PR TITLE
ES6

### PR DESCRIPTION
**Pull request by BLamy**
Here are some suggestions for getting started with es6.

- [x] Add a yarn.lock
- [x] use airbnb's eslint rules
- [x] transpile with babel es2015 preset
- [x] use babel module resolver instead of requireish
- [ ] use [lebab](https://github.com/lebab/lebab) if possible

### CODE REVIEW

**Comment by BLamy** [comment]
Even if the library never moves to ES6 I still think the babel module resolver should be used over requireish. requireish has burned me several times. 

image

File: test/transport/test_client_tcp_transport.js [hunk 1]
```diff
@@ -177,7 +177,7 @@ describe("testing ClientTCP_transport", function () {
         });
         fake_server.pushResponse(spyOnServerWrite);
 
-        transport.timeout = 10; // very short timeout;
+        transport.timeout = 100; // very short timeout;
```
Comment (BLamy) [inline test/transport/test_client_tcp_transport.js:180]: The comment makes me think there was a reason it was only at 10.

Comment (gilesbradshaw) [reply test/transport/test_client_tcp_transport.js:180]: you might well be right - I think @erossignon wld need to look at that..

Comment (gilesbradshaw) [reply test/transport/test_client_tcp_transport.js:180]: ah ok - now passing on Node 6+  :)

**Comment by BLamy** [comment]
@gilesbradshaw 

A little confused as to why node 5 completely crapped out. 
Node 4 is just missing a polyfill. 

.12 Issues can probably be resolved but I think an argument could be made for dropping it all together.

**Comment by gilesbradshaw** [comment]
I've been playing with this and have ES6ed opcua-client and client_base and then started lebabing the various transforms.  However it seems to be the tests are not completely determinate.  I now get errors even when I go back to the revision I have PRed to you....

**Comment by gilesbradshaw** [comment]
Not sure if tests are totally robust - I am now getting #298 even on original...

**Comment by BLamy** [comment]
I'm going to hold off on merging your [giant PR](https://github.com/BLamy/node-opcua/pull/3) until we hear from @erossignon. I want to make sure he agrees with the opinions we've taken thus far before I overwhelm him with one giant PR.

**Comment by gilesbradshaw** [comment]
no problem - good idea!

**Comment by erossignon** [comment]
Hi, I am investigating how we could bring @gilesbradshaw's ES6 branch that is now compiling OK into a 'ES6-experimental branch" of the official node-opcua repository , so we could keep going together .
Not quite sure yet, what is the best route: Giles has forked Bret's fork, so I cannot merge Giles' PR directly I guess.

**Comment by BLamy** [comment]
Looks like the es6 branch got published.

Closing in favor of tracking progress here:
https://github.com/node-opcua/node-opcua/issues/292

**Pull request closed without merging**